### PR TITLE
init docs

### LINKED ISSUE #468: Glorot gains for Leaky_rectify and very_leaky_rectify
Should we add the correct Glorot gains for `leaky_rectify` and `very_leaky_rectify`. That would be

```Python
if gain == 'rectify':
    gain = np.sqrt(2)
elif nonlin == 'leaky_rectify':
    gain = np.sqrt(2/(1+0.01**2))
elif nonlin == 'very_leaky_rectify':
    gain = np.sqrt(2/(1+(1/3)**2))
```

Maybe also add a note to the doc that the correct  init for `leaky_rectify` with `alpha` leakiness is `np.sqrt(2/(1+alpha**2))`

### PR DESCRIPTION
**Pull request by skaae**
fixes #468 
Also add toc to inits. Makes it easier to read. Let me know what you think? 

- use ‘relu’ keyword in example. I think that is the most typical usecase?
- add autosummary to init docs
- add leaky relu formula to glorot
- moves some references that does not work in autosummary

### CODE REVIEW

File: lasagne/init.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 --------
 >>> from lasagne.layers import DenseLayer
 >>> from lasagne.init import Constant, GlorotUniform
->>> l1 = DenseLayer((100,20), num_units=50, W=GlorotUniform(), b=Constant(0.0))
+>>> l1 = DenseLayer((100,20), num_units=50, W=GlorotUniform('relu'), b=Constant(0.0))
```
Comment (f0k) [inline lasagne/init.py]: Line too long! For a multiline doctest, the format is:
```python
>>> l1 = DenseLayer(something something something
...                 something something)
```

File: lasagne/init.py [hunk 2]
```diff
@@ -100,9 +100,9 @@ def sample(self, shape):
 
 
 class Glorot(Initializer):
-    """Glorot weight initialization [1]_.
+    """Glorot weight initialization.
 
-    This is also known as Xavier initialization.
+    This is also known as Xavier initialization[1]_.
```
Comment (f0k) [inline lasagne/init.py]: +1 for moving the reference (to fix the autosummary), but you're missing a blank before the `[1]_`.

File: lasagne/init.py [hunk 3]
```diff
@@ -141,6 +141,15 @@ class Glorot(Initializer):
        \\sigma &= \\sqrt{\\frac{2}{fan_{in}+fan_{out}}}\\\\
        W &\sim N(0, \\sigma)
 
+    For leaky ReLU ``gain`` can be calculated as
+
+    .. math::
+       gain = \\sqrt{\\frac{2}{1+\\alpha^2}}
+
+    where :math:`\\alpha` is the leakyness. For
```
Comment (f0k) [inline lasagne/init.py]: leakyness -> leakiness

File: lasagne/init.py (hunk 1 above)
Comment (skaae) [inline lasagne/init.py]: noob mistanke. sorry about that :)

File: lasagne/init.py [hunk 4]
```diff
@@ -193,10 +202,10 @@ def __init__(self, gain=1.0, c01b=False):
 
 
 class He(Initializer):
-    """He weight initialization [1]_.
+    """He weight initialization.
 
     Weights are initialized with a standard deviation of
-    :math:`\\sigma = gain \\sqrt{\\frac{1}{fan_{in}}}`.
+    :math:`\\sigma = gain \\sqrt{\\frac{1}{fan_{in}}}`. [1]_
```
Comment (skaae) [inline lasagne/init.py]: Moved the dot here as well.

File: lasagne/init.py [hunk 5]
```diff
@@ -141,6 +141,15 @@ class Glorot(Initializer):
...
        W &\sim N(0, \\sigma)
 
+    For leaky ReLU ``gain`` can be calculated as
+
+    .. math::
+       gain = \\sqrt{\\frac{2}{1+\\alpha^2}}
+
+    where :math:`\\alpha` is the leakyness. For
+    ``leaky_rectify`` and
+    ``very_leaky_rectify`` :math:`\\alpha` is  0.01 and 0.333 respectively.
```
Comment (f0k) [inline lasagne/init.py]: Can you correctly cross-reference those? Should be:
```
:func:`leaky_rectify <lasagne.nonlinearities.leaky_rectify>`
```
or shorter (never tried that):
```
:func:`~lasagne.nonlinearities.leaky_rectify`
```

Comment (skaae) [inline lasagne/init.py]: Great. Tried to do cross-reference but i couldn't figure out how to reference a function.

**Comment by f0k** [comment]
Actually, wouldn't it be easier to just give the formula in the description of the `gain` parameter? We currently have:
```
     Scaling factor for the weights. Set this to 1.0 for linear and sigmoid
     units, to 'relu' or sqrt(2) for rectified linear units. Other transfer
     functions may need different factors.
```
This could just become:
```
    Scaling factor for the weights. Set this to ``1.0`` for linear and sigmoid
    units, to 'relu' or ``sqrt(2)`` for rectified linear units, and to
    ``sqrt(2/(1+alpha**2)`` for leaky rectified linear units with leakiness
    ``alpha``. Other transfer functions may need different factors.
```
That's shorter than your "Notes" paragraph and could be copied over to all initializers that have a gain factor.

**Comment by skaae** [comment]
Yes thats fine with me. Is the gain the same for Glorot and He?

**Comment by skaae** [comment]
+ for orthogonal?

**Comment by f0k** [comment]
It's the same for all three. While you're at it, can you reference http://arxiv.org/abs/1312.6120 in the orthogonal initialization?

**Comment by skaae** [comment]
yes

**Comment by f0k** [comment]
Cool! Looks good to merge for me!

**Comment by f0k** [comment]
Since there are no objections, I'll merge it. Thanks again!

**Pull request merged**